core/block/object/objectlink/dependent_objects.go

```diff
@@ -74,65 +68,133 @@ func DependentObjectIDs(s *state.State, converter KeyToIDConverter, blocks, deta
 			ids = append(ids, id)
 		}
 
-		if !details {
+		if !flags.Details {
 			continue
 		}
 
-		// handle corner cases first for specific formats
-		if rel.Format == model.RelationFormat_date &&
-			!lo.Contains(bundle.LocalAndDerivedRelationKeys, rel.Key) {
-			relInt := pbtypes.GetInt64(det, rel.Key)
-			if relInt > 0 {
-				t := time.Unix(relInt, 0)
-				t = t.In(time.Local)
-				ids = append(ids, addr.TimeToID(t))
+		ids = append(ids, collectIdsFromDetail(rel, det, flags)...)
+	}
+
+	if flags.Collection {
+		ids = append(ids, s.GetStoreSlice(template.CollectionStoreKey)...)
+	}
+
+	ids = lo.Uniq(ids)
+	return
+}
+
+func collectIdsFromBlocks(s *state.State, flags Flags) (ids []string) {
+	err := s.Iterate(func(b simple.Block) (isContinue bool) {
+		if flags.DataviewBlockOnlyTarget {
+			if dv := b.Model().GetDataview(); dv != nil {
+				if dv.TargetObjectId != "" {
+					ids = append(ids, dv.TargetObjectId)
+				}
+				return true
 			}
-			continue
 		}
 
-		if rel.Key == bundle.RelationKeyCreator.String() ||
-			rel.Key == bundle.RelationKeyLastModifiedBy.String() {
-			if creatorModifierWorkspace {
-				v := pbtypes.GetString(det, rel.Key)
-				ids = append(ids, v)
+		if flags.NoImages {
```

Comment:
what if this flag is false?

Reply:
If flag is false, then file block will be processed by FillSmartIds logic, because file block implements linkSource interface

Reply:
I suggest you to leave a comment about this

Reply:
Done!